expansions/pandemonium/src/main/java/ladysnake/pandemonium/common/entity/ability/WitherSkullAbility.java

```diff
@@ -0,0 +1,133 @@
+package ladysnake.pandemonium.common.entity.ability;
+
+import ladysnake.requiem.common.entity.ability.DirectAbilityBase;
+import ladysnake.requiem.common.entity.ability.IndirectAbilityBase;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.entity.boss.WitherEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.projectile.WitherSkullEntity;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.math.Vec3d;
+
+import java.util.Random;
+
+public class WitherSkullAbility {
+    public static class BlueWitherSkullAbility extends IndirectAbilityBase<WitherEntity> {
```

Comment:
I would rather you make `WitherSkullAbility` extend `AbilityBase`, put all the common methods inside, then make both implementations extend `WitherSkullAbility`, with `BlueWitherSkullAbility` implementing `IndirectAbility` and `BlackWitherSkullAbility` implementing `DirectAbility`. This would reduce the amount of duplicated code quite a bit.

Reply:
Alright, I'll make those changes

Reply:
done